src/commands/sign.ts

```diff
@@ -1,69 +1,44 @@
 import { Argv } from "yargs";
+import { withWalletOption } from "./shared";
 import signale from "signale";
-import { sign, Output } from "../implementations/sign";
-import { isDir } from "../implementations/utils/disk";
+import { sign, SignCommand, supportedAlgorithms } from "../implementations/sign";
 
 export const command = "sign <raw-documents-path>";
 
 export const describe = "Sign document(s) and appends proof block";
 
 export const builder = (yargs: Argv): Argv =>
-  yargs
-    .positional("raw-documents-path", {
-      description: "Directory containing the unissued raw documents or a single raw document file",
-      normalize: true,
-      type: "string",
-    })
-    .option("output-file", {
-      alias: "of",
-      description: "Write output to a file. Only use when <wrapped-documents-dir> is a document",
-      type: "string",
-      conflicts: "output-dir",
-    })
-    .option("output-dir", {
-      alias: "od",
-      description: "Write output to a directory",
-      type: "string",
-      conflicts: "output-file",
-    })
-    .option("private-key", {
-      description: "Private key to sign document(s) with",
-      type: "string",
-    })
-    .option("public-key", {
-      description: "Public key, added to the document, used to verify the signature is valid",
-      type: "string",
-    });
-
-export interface SignCommand {
-  rawDocumentsPath: string;
-  outputDir?: string;
-  outputFile?: string;
-  privateKey: string;
-  publicKey: string;
-}
+  withWalletOption(
+    yargs
+      .positional("raw-documents-path", {
+        description: "Directory containing the unissued raw documents or a single raw document file",
+        normalize: true,
+        type: "string",
+      })
+      .option("output-dir", {
+        alias: "od",
+        description: "Write output to a directory",
+        type: "string",
+        required: true,
+      })
+      .option("public-key", {
+        alias: "p",
+        description: "Public key in did (ie did:ethr:0xE712878f6E8d5d4F9e87E10DA604F9cB564C9a89#controller)",
+        type: "string",
+        required: true,
+      })
+      .option("algorithm", {
```

Comment:
any plan to add an option to sign and wrap ? =)

Reply:
Nope, in fact I disagree that the wrapping command is doing unwrap too, i want to make another PR to do the unwrapping separately (find myself needing that quite often). 

I like how CLI should do one thing and do one thing well.... If needed they can be composed together.

Reply:
the composition is not very simple and easy to use at all